### ISSUE TITLE
[HOLD for payment 2025-02-06] [HOLD for payment 2025-01-21] Add plan changes to Subscriptions 

### ISSUE THREAD

**Issue by garrettmknight** [Expensify/App#52588]
# Part of the Workspace Downgrades project
Main issue: https://github.com/Expensify/Expensify/issues/399936
Project: Workspace Downgrades

# Feature Description
1. Update the `Your plan` section of the Subscriptions page to show both plan types with the current, highest plan type the account is an admin of highlighted/selected.
2. If the user selects the other plan type, we'll check whether they're an admin on one or more workspaces:
- If they're only an admin of one workspace we can just upgrade or downgrade that workspace from here we'll:
  - If the user is on Collect and selects Control, we'll launch the Upgrade RHP
  - If the user is on Control and selects Collect, we'll launch the Downgrade RHP
- If they're an admin on multiple workspaces, they'll need to upgrade or downgrade their workspaces individually:
  - If the user is on Collect and selects Control, we'll launch an RHP directing the user to workspaces to upgrade
  - If the user is on Control and selects Collect, we'll launch an RHP directing the user to workspaces to downgrade

Copy for options above in this **[Figma here](https://www.figma.com/design/MetiE9RmxrhHeJ1ZXDGMrV/Workspace-Downgrades%2FUpgrades?node-id=4075-9912&node-type=canvas&t=cRCIX9gxA9DBAYrX-0)**

# Manual Test Steps

**Single Workspace Upgrade**
1. Create a new account
2. Create a workspace
3. Navigate to subscriptions
4. Click on 'Control'
5. Confirm Upgrade RHP opens
6. Click 'Upgrade'
7. Confirm your workspace has been upgraded

**Single Workspace Downgrade (follow Single Workspace Upgrade steps first)**
1. Navigate to subscriptions
2. Click on 'Collect'
3. Confirm Downgrade RHP opens
4. Click 'Downgrade'
5. Confirm your workspace has been downgraded

**Multi Workspace Upgrade (follow Single Workspace Upgrade + Downgrade steps first)**
1. Create an additional workspace
2. Navigate to subscriptions
3. Click on 'Control'
4. Confirm Upgrade RHP opens for multi workspaces
5. Click 'Go to Workspaces'
7. Confirm you're sent to workspace page
8. Upgrade a workspace
9. Navigate back to Subscriptions and confirm 'Control' is selected

**Multi Workspace Upgrade (follow Multi Workspace Upgrade steps first)**
1. Navigate to subscriptions
2. Click on 'Collect'
3. Confirm Downgrade RHP opens for multi workspaces
4. Click 'Go to Workspaces'
5. Confirm you're sent to workspace page
6. Downgrade the Control workspace
7. Navigate back to Subscriptions and confirm 'Collect' is selected

# Automated Tests
N/A

<details><summary>Issue Owner</summary>Current Issue Owner: @garrettmknight</details>

**Comment by JmillsExpensify** [comment]
Still very excited for this one.

**Comment by garrettmknight** [comment]
Hoping to prioritize this week while things are quiet.

**Comment by garrettmknight** [comment]
Alright - OP is updated, now we're just waiting for the main issue to complete to open this one up.

**Comment by garrettmknight** [comment]
**Quick clarification about the HOLD here:**
- We already have the upgrade command in ND
- We're working on the downgrade command in https://github.com/Expensify/Expensify/issues/399936
- Once we have the downgrade command, we should be able to open this up externally since it's more of a FE change that just calls our BE command.

**Comment by carlosmiceli** [comment]
@jayeshmangwani Hey Jayesh, here's another issue we need to work on to really complete the full upgrades/downgrades project. Seems like a natural progression for you to take over this issue too, let me know if there are any questions.

I do have one question for you: we need to implement a check for whether the admin manages multiple workspaces or not. Do we have a way to check for that in the FE? Probably not, but thought I'd ask. If not, what kind of changes would you need from the BE (as in, what command or response would you like to see changed and how)?

**Comment by jayeshmangwani** [comment]
> Seems like a natural progression for you to take over this issue too, let me know if there are any questions.

Yes, that sounds good. I will start working on this issue once we merge PR for the downgrade workspace UI and new downgrade API. 

@carlosmiceli @garrettmknight I will also need access to Figma, as I cannot find this section in the design doc.

**Comment by carlosmiceli** [comment]
Boom! Looks like we're good here @garrettmknight 💪 
@jayeshmangwani great news, we're waiting for the final BE review and we're good to go, will report as soon as it's merged 👍

**Comment by jayeshmangwani** [comment]
> will report as soon as it's merged 👍

Thanks 🙌

**Comment by carlosmiceli** [comment]
Small BE change requested by our contributors has both PRs in review, which should completely unblock @jayeshmangwani from completing the plan page work.

**Comment by carlosmiceli** [comment]
Both BE PRs have been merged.

**Comment by jayeshmangwani** [comment]
Not Overdue, we will handle it after this [PR](https://github.com/Expensify/App/pull/54356)

**Comment by jayeshmangwani** [comment]
@garrettmknight Whenever you get a chance, please help in get access to the Figma design/ Screenshots here

**Comment by carlosmiceli** [comment]
Not overdue, it can now be worked on since the downgrade feature PR ahs been merged.

**Comment by jayeshmangwani** [comment]
Since we can now work on these changes, @carlosmiceli could you please provide the UI screenshots that need to be implemented?

**Comment by carlosmiceli** [comment]
To be honest I'm not sure where those are, let's wait for @garrettmknight to provide them.

**Comment by carlosmiceli** [comment]
@jayeshmangwani 

Single Upgrade:

![image](https://github.com/user-attachments/assets/b9fcf621-31d7-4ac3-b15a-31b8356534d9)

![image](https://github.com/user-attachments/assets/9a311f1c-cb55-4bb1-b0dd-fe6c0daf7e18)

Multi Upgrade:

![image](https://github.com/user-attachments/assets/0ae7e8c6-3bee-41f6-83b8-40c40c262c69)

![image](https://github.com/user-attachments/assets/187ec6ad-4d20-4263-b406-0880fa95a211)


Single Downgrade:

![image](https://github.com/user-attachments/assets/08b76be1-96ae-41c3-8572-6a66306abb90)

![image](https://github.com/user-attachments/assets/242d87cf-e911-4191-bc21-2d9e6273bac9)

Multi Downgrade:

![image](https://github.com/user-attachments/assets/dbcddbaf-c190-4f80-9144-34fb63f6540e)

![image](https://github.com/user-attachments/assets/9e549334-d44a-4ba9-848a-8faf19869e3b)

**Comment by garrettmknight** [comment]
@jayeshmangwani just a note that when the RHP is up, there should be a shadow on the rest of the screen.

**Comment by jayeshmangwani** [comment]
Thanks for the mockups, @carlosmiceli! I’ll start working on the PR today.

**Comment by jayeshmangwani** [comment]
@carlosmiceli , @garrettmknight ,New plan type design looks good on web (screenshot added below for comparison with other devices).
<details>
  <summary>web: chrome</summary>
  
<img width="1512" alt="Screenshot 2024-12-31 at 9 50 25 PM" src="https://github.com/user-attachments/assets/2eb83bea-f346-4243-8228-4751ca27aac9" />

</details>
However, on native mobile devices and narrow web layouts, it doesn't look as polished. I've attached screenshots from iOS and Safari on mobile web for reference.

Should we update the layout to display the plan type in a column format for narrow layouts? Or would it be better to get input from the design team on this?
<details>
  <summary>iOS</summary>
  
![Simulator Screenshot - iPhone 15 Pro Max - 2024-12-31 at 21 50 03](https://github.com/user-attachments/assets/c401477a-28a4-4584-9c68-5ec14944947a)


</details>
<details>
  <summary>mweb: safari</summary>
  
![Simulator Screenshot - iPhone 15 Pro Max - 2024-12-31 at 21 49 31](https://github.com/user-attachments/assets/ddc48e12-f892-49c7-8814-32c75c6ea4fd)

</details>




  I tried using the column layout for mobile devices, but that doesn’t look great either. Since the Plan type cards have some height, they end up taking over the whole screen. I've attached a screenshot below for reference.

<details>
  <summary>iOS: column flex direction</summary>
  
![Simulator Screenshot - iPhone 15 Pro Max - 2024-12-31 at 21 55 12](https://github.com/user-attachments/assets/40b34fa3-0ff5-4555-86d6-65a2a53bc2c2)


</details>

**Comment by carlosmiceli** [comment]
Oh, sorry, I think this should be @DylanDylann as the C+ reviewer.

**Comment by carlosmiceli** [comment]
But let's wait for @garrettmknight's input on the UI above.

**Comment by shawnborton** [comment]
cc @Expensify/design for a review of the UI above

I think the idea was that on mobile, these would be stacked (aka flex column). The web screenshots look good to me though.

**Comment by garrettmknight** [comment]
Just catching up from NY. I like the stacked option, let's go with that.

**Comment by dubielzyk-expensify** [comment]
Agree with the mobile stacking Shawn called out. Is it just me or does this look a bit cramped here? Is that our regular line-height? If so, maybe we should add just a tad more padding between the plan price paragraph and the name of the plan itself? 
![CleanShot 2025-01-06 at 15 11 34@2x](https://github.com/user-attachments/assets/748a259f-6acd-4ea1-a8ff-61bcd39c846c)

**Comment by jayeshmangwani** [comment]
> Is that our regular line-height?

Yes, Line height is not being set explicitly, so the text is using the default font size's normal line height (20px). The font size for the title is 15px, and its corresponding line height is 20px.

@dubielzyk-expensify I’ve added screenshots below for iOS and web after applying an 8px margin between the title and description text. Please take a look.
<details>
  <summary>web</summary>
  
<img width="1512" alt="Screenshot 2025-01-06 at 11 15 16 AM" src="https://github.com/user-attachments/assets/eaeb575b-add7-4ae0-9a14-e00a4869a541" />

  
</details>

<details>
  <summary>iOS</summary>
  
![Simulator Screenshot - iPhone 15 Pro Max - 2025-01-06 at 11 15 04](https://github.com/user-attachments/assets/ba379c44-53b2-4e49-a3c5-c019fdd2bfcf)

</details>

**Comment by dannymcclain** [comment]
100% agree about the mobile stack. And I like the addition of that 8px of space in the screenshots above. Looking good to me.

**Comment by garrettmknight** [comment]
Agreed that the most recent screenshots look great - LGTM!

**Comment by dubielzyk-expensify** [comment]
Awesome! Thanks for the swift fix @jayeshmangwani

**Comment by shawnborton** [comment]
Dropping by to say the same - I like the addition of that 8px margin, looking good!

**Comment by carlosmiceli** [comment]
Back from OOO for a week, thanks everyone for finalizing the details here! PR is merged :)

**Comment by jayeshmangwani** [comment]
@garrettmknight, there is no payment for this issue. We will handle it at the end when the project's payment will be settled, We can close this issue

**Comment by DylanDylann** [comment]
@jayeshmangwani Have the project finished? I think we already completed all tasks

**Comment by jayeshmangwani** [comment]
@DylanDylann Yes, all major tasks are completed. However, minor copy changes need to be made on the Plan Type page to indicate that users must change their plan one by one for policies if user have multiple workspaces as an admin. However, this issue has not been created anywhere yet, though there was some discussion about it on Slack last week.

> Have the project finished? 

For this, ccing @garrettmknight @carlosmiceli

**Comment by garrettmknight** [comment]
Just asked for a second set of eyes on copy [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1737640082467279?thread_ts=1731682677.445059&cid=C02NK2DQWUX).

**Comment by garrettmknight** [comment]
Alright, agreed upon copy for the downgrade RHP from the Subscriptions page when the user owns more than one workspace:

```
Some features are only available on the Control plan. If you downgrade, you’ll lose access to these features and more:
- Accounting connections (except QBO or Xero)
- Expense rules
- Multiple-level approval workflows
- Enhanced security controls

For a full feature list and comparison of our plans, check out our [pricing page](https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/Plan-types-and-pricing).

**Heads up!** You’ll need to downgrade all your workspaces before your first monthly payment to begin a subscription at the Collect rate. Click **Go to Workspaces** > select each workspace > change the plan type to **Collect**.
```

**Comment by jayeshmangwani** [comment]
> can you make that update as part of this issue?

 Sure thing, I'll open a new PR for the changes [mentioned](https://github.com/Expensify/App/issues/52588#issuecomment-2610459137) above.

**Comment by jayeshmangwani** [comment]
@garrettmknight, currently, we show the text `Are you sure you want to downgrade and remove your configurations? **This cannot be undone**.` above the 'Go to Workspaces' button. Should we hide it in the case of multiple workspaces and only show it for a single workspace, or should we remove this text altogether?

<img width="375" alt="Image" src="https://github.com/user-attachments/assets/f3fe6855-fec9-43dc-ac21-dd8a0c725328" />

**Comment by garrettmknight** [comment]
I think we can remove the 'this cannot be undone' warning - they'll see it when they downgrade the workspaces individually. 

Here's the upgrade copy:

```
Unlock our most powerful features, including:
- Advanced accounting connections (NetSuite, Sage Intacct, and more)
- Smart expense rules
- Multi-level approval workflows
- Enhanced security controls

The Control plan starts at $9 per active member per month. Learn more about our [plans and pricing](https://help.expensify.com/articles/new-expensify/billing-and-subscriptions/Plan-types-and-pricing).


To upgrade, click **Go to Workspaces**, select a workspace, and change the plan type to **Control**.
```

**Comment by jayeshmangwani** [comment]
Thanks for the copies

**Comment by garrettmknight** [comment]
@jayeshmangwani let me know when you've got that PR up.

**Comment by jayeshmangwani** [comment]
@garrettmknight @DylanDylann Opened a [PR](https://github.com/Expensify/App/pull/55858) to update the copies, but we still need confirmation for the Spanish version, which we raised on Slack.

**Comment by garrettmknight** [comment]
No BZ necessary, adding QA for the project.

**Comment by DylanDylann** [comment]
@garrettmknight @carlosmiceli If there are no tasks anymore. I think we are ready to handle the payment for this entire project.

cc @jayeshmangwani

**Comment by jayeshmangwani** [comment]
Agreed with Dylan—there are no pending tasks for this project.

**Comment by DylanDylann** [comment]
We have 6 issues created for this feature and all are done by 4 PRs. Additionally we have some other regression/polish issues that not calculated here

1. https://github.com/Expensify/App/issues/51486
2. https://github.com/Expensify/App/issues/52588
3. https://github.com/Expensify/App/issues/51768
4. https://github.com/Expensify/App/issues/51494
5. https://github.com/Expensify/App/issues/51764
6. https://github.com/Expensify/App/issues/51515

**Comment by DylanDylann** [comment]
@garrettmknight Bump on [this comment](https://github.com/Expensify/App/issues/52588#issuecomment-2635699646)

**Comment by garrettmknight** [comment]
Thanks @DylanDylann - "Additionally we have some other regression/polish issues that not calculated here" Where are those other issues?

**Comment by garrettmknight** [comment]
Collecting everything together here: https://github.com/Expensify/App/issues/56677
